### DISCUSSION TITLE
Problem Executing Command with sqlpage.exec() in SQLPage

### DISCUSSION THREAD

**Discussion by sijucj** [sqlpage/SQLPage#685] [General]
**Observations**: The following commands work as expected:

```
SELECT 'debug' as component; 
SELECT sqlpage.exec('surveilr', '--version') as version;
```

However, the following commands fails:
```
SELECT 'debug' as component; 
SELECT sqlpage.exec('surveilr ingest files -r ' || sqlpage.current_working_directory() || '/study-files/', '');
SELECT 'debug' as component; 
SELECT sqlpage.exec('surveilr', 'ingest files -r ' || sqlpage.current_working_directory() || '/study-files/');
SELECT 'debug' as component; 
SELECT sqlpage.exec('surveilr ingest', 'files -r ' || sqlpage.current_working_directory() || '/study-files/');
```

![image](https://github.com/user-attachments/assets/81817cde-1481-4151-a163-f1b21998d4e5)

**Description:**
I am trying to execute a command using sqlpage.exec() to run the Surveilr ingest process, but it fails.
The goal is to execute the Surveilr ingestion process with a dynamic directory path using SQLPage.

**Environment:**
SQLPage Version: 0.30.1
Surveilr Version: 1.1.0

**Questions:**
Why does these commands fail?

Any insights or suggestions to resolve this issue would be greatly appreciated.

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!

The root cause of the issue is that `sqlpage.exec` requires each CLI argument to be passed as a separate parameter. When you concatenate multiple arguments into a single string, sqlpage.exec() interprets this as one argument rather than separate ones.

To fix this, split the command into separate arguments within sqlpage.exec(). Here’s how you can modify the commands:

```sql
SELECT 'debug' as component;
SELECT sqlpage.exec('surveilr', 'ingest', 'files', '-r', sqlpage.current_working_directory() || '/study-files/');
```

Each part of the command (`'ingest'`, `'files'`, `'-r'`, and the directory path) is now passed as a separate argument, allowing sqlpage.exec() to execute the command correctly.

**Reply by sijucj** [reply]
Thank you so much for your help in fixing the issue!  Your solution worked perfectly, and I really appreciate the time you took to assist me
